src/main/java/uk/gov/hmcts/reform/sscs/service/LetterUtils.java

```diff
@@ -197,4 +191,112 @@ public static Stream<Pair<String, YesNo>> buildOtherPartiesForReasonableAdjustme
         }
         return otherPartyReasonableAdjustmentList.stream();
     }
+
+    public static String getNameForSender(SscsCaseData sscsCaseData) {
+        if (nonNull(sscsCaseData.getOriginalSender())) {
+            final String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
+            if (originalSenderCode.equalsIgnoreCase("appellant")) {
+                return sscsCaseData.getAppeal().getAppellant().getName().getFullNameNoTitle();
+            } else if (originalSenderCode.equalsIgnoreCase("representative")) {
+                return SendNotificationHelper.getRepSalutation(sscsCaseData.getAppeal().getRep(), false);
+            } else if (originalSenderCode.equalsIgnoreCase("jointParty")) {
+                return sscsCaseData.getJointParty().getName().getFullNameNoTitle();
+            } else {
+                return getOtherPartyName(sscsCaseData)
+                        .map(Name::getFullNameNoTitle).orElse("");
+            }
+        }
+        return "";
+    }
+
+    private static Optional<Representative> getRepresentativeOfOtherParty(SscsCaseData sscsCaseData) {
+        for (CcdValue<OtherParty> op : sscsCaseData.getOtherParties()) {
+            if (op.getValue().hasRepresentative() && sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getRep().getId())) {
+                return Optional.of(op.getValue().getRep());
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Name> getOtherPartyName(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOriginalSender().getValue().getCode().contains("otherPartyRep")) {
```

Comment:
extract if condition as local variable

Reply:
Fixed.

Reply:
I don't see a change. I was expecting
```suggestion
        boolean isOtherPartyRep = sscsCaseData.getOriginalSender().getValue().getCode().contains("otherPartyRep");
        if (isOtherPartyRep) {
```

Reply:
Updated.